sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,479 @@
+
+        # Update the tags by iterating through all tags
+        for tag in true_if_any_tags:
+            # Checking the equivalent forecaster tags
+            true_if_any = True
+
+            for forecaster in self.forecasters.values():
+                # Fetching the forecaster tags
+                forecaster_tags = forecaster.get_tags()
+
+                if tag in forecaster_tags and forecaster_tags[tag] is True:
+                    true_if_any = True
+                    break
+
+            # Perform this check for the fallback forecaster too
+            if (
+                fallback_forecaster is not None
+                and tag in fallback_forecaster.get_tags()
+                and fallback_forecaster.get_tags()[tag] is True
+            ):
+                true_for_all = True
+
+            true_if_any_tags[tag] = true_if_any
+
+        # Update the tags
+        self.set_tags(**true_if_all_tags)
+        self.set_tags(**true_if_any_tags)
+
+        # Finally, dynamically adding implementation of probabilistic
+        # functions depending on the tags set. This is important since
+        # having an implementation of the the _predict_intervals function
+        # when the relevant tags are False (which, in turn, are set solely
+        # by the passed forecasters) leads to the test_pred_int_tag test to fail.
+
+        if self.get_tags()["capability:pred_int"]:
+            self._update_predict_single = _update_predict_single
+            self._predict_interval = _predict_interval
+            self._predict_var = _predict_var
+            self._predict_proba = _predict_proba
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : Pd.Series
+            The target time series to which we fit the data.
+
+        fh : ForecastingHorizon | None, optional (default=None)
+            The forecasting horizon with the steps ahead to predict.
+
+        X : Pd.Series | None, optional (default=None)
+            No exogenous variables are used for this.
+
+        Returns
+        -------
+        self : reference to self
+
+        Raises
+        ------
+        ValueError: If the extrapolated category has no provided forecaster
+        and if there is no fallback forecaster provided to the object!
+
+        Example:
+
+        If the passed transformer is an ADICVTransformer(), and the generated
+        series is a lumpy series; however, if there is no key matching "lumpy"
+        in the forecasters parameter, the fallback_forecaster will be used.
+        Additionally, if the fallback_forecaster is None, a ValueError will be thrown.
+        """
+        # passing time series through the provided transformer!
+
+        self.category_ = self.transformer_.fit_transform(X=y).iloc[0, 0]
```

Comment:
this should be explained in the docstring, i.e., that the first column is used.

Or, do we want to support multiple columns? In which case the keys could perhaps be tuples? You can get these via `itertuples`, and the length 1 case needs special treatment.

Reply:
We could do it that way. However, would it then be a requirement that the first row must have the key that we are looking for? Also, if we are doing multiple columns (by which I assume you mean that the key could be in any column, but it would be in only in one specific column) would it be simpler if the user could pass an additional column name as a parameter where we should search for the key?

I'm okay with either design choice (user possibly specifies which column to choose or sets up the transformer such that the first column is the valid one), but could you tell me which approach might be easiest for the user?

Reply:
I would say, let's use the first column for now and worry about more columns later.